### DISCUSSION TITLE
Optional[list] is not working with callbacks

### DISCUSSION THREAD

**Discussion by solesensei** [fastapi/typer#762] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
import typing as tp

app = typer.Typer(name="abc")

def callback(
        ctx: typer.Context,
        param: typer.CallbackParam,
        value: tp.List[str] | None
) -> tp.List[str] | None:
    return value


@app.command("test", help="Test")
def test_cmd(
    abc: tp.Optional[tp.List[str]] = typer.Option(
        None, "--abc", "-a", help="abc", show_default=False,
        callback=callback,
    ),
) -> None:
    print(abc)


@app.callback()
def main():
    pass

if __name__ == "__main__":
    app()
```

### Description

Related https://github.com/tiangolo/typer/issues/410 and https://github.com/tiangolo/typer/pull/664

```
python main.py test
site-packages/typer/main.py:637 in internal_convertor
TypeError: object of type 'NoneType' has no len()
```

`Optional[list] = Option(None, callback=callback)` not working with None's anymore.
On `v0.9.0` – everything works.

### Operating System

macOS

### Typer Version

0.12.3

### Python Version

Python 3.11.8

**Comment by solesensei** [comment]
Can reproduce on 12.5 version

**UPD: the fix is here https://github.com/fastapi/typer/pull/1018**

**Reply by solesensei** [reply]
@tiangolo can you check?

**Comment by sfc-gh-jvasquezrojas** [comment]
I am experiencing the same issue. The problems occurs when the default value converter `generate_list_convertor` is executed. It does not validate if the input value is None before checking if `len(value) == 0`. This issue is specifically found in the following line: [GitHub link](https://github.com/tiangolo/typer/blob/8ab3a3534ae1423b5e97d7f4ab44b9ac6257d1a9/typer/main.py#L639C9-L639C54).

**Comment by solesensei** [comment, marked as answer]
Fix is here https://github.com/fastapi/typer/pull/1018
